Examples/javascript/native/example.i

```diff
@@ -0,0 +1,104 @@
+/* File : example.i */
+%module example
+
+
+%wrapper
+%{
+#include <cstring>
+#include <string>
+
+
+#if (V8_MAJOR_VERSION-0) < 4 && (SWIG_V8_VERSION < 0x031903)
+  typedef v8::Handle<v8::Value> V8ReturnValue;
+  typedef v8::Arguments V8Arguments;
+# define V8_RETURN(val) return scope.Close(val)
+#else
+  typedef void V8ReturnValue;
+  typedef v8::FunctionCallbackInfo<v8::Value> V8Arguments;
+# define V8_RETURN(val) args.GetReturnValue().Set(val); return
+#endif
+#if (V8_MAJOR_VERSION-0) < 4 && (SWIG_V8_VERSION < 0x032318)
+#    define V8_UNDEFINED() v8::Undefined()
+#else
+#    define V8_UNDEFINED() v8::Undefined(v8::Isolate::GetCurrent())
+#endif
+#if (V8_MAJOR_VERSION-0) < 4 && (SWIG_V8_VERSION < 0x032117)
+#    define V8_HANDLESCOPE() v8::HandleScope scope
+#elif (V8_MAJOR_VERSION-0) < 4 && (SWIG_V8_VERSION < 0x032224)
+#    define V8_HANDLESCOPE() v8::HandleScope scope(v8::Isolate::GetCurrent());
+#else
+#    define V8_HANDLESCOPE() v8::HandleScope scope(v8::Isolate::GetCurrent());
+#endif
+
```

Comment:
Did you see the macros in javascriptrun.swg which provide SWIGV8_HANDLESCOPE, SWIGV8_RETURN? Probably better to just use those instead.

Reply:
Good idea! I actually was aware of those, but I was fence-sitting on using them. The idea of tightly coupling objects.

I will update them accordingly!